napari/utils/colormaps/colormaps.py

```diff
@@ -111,6 +118,36 @@ def _low_discrepancy_image(image, seed=0.5, margin=1 / 256):
     return image_out
 
 
+def color_dict_to_colormap(color_dict):
+    """
+    Generate a color map based on the given color dictionary
+    Parameters
+    ----------
+    color_dict : dict of int
+        Mapping between labels and color strings
+
+    Returns
+    -------
+    colormap : Colormap
+        Colormap constructed with provided control colors
+    label_color_index : dict of int
+        Mapping of Label to color control point within colormap
+    """
+    colors = [
+        transform_color(color_str)[0]
+        for label, color_str in color_dict.items()
+    ]
+
+    colormap = Colormap(colors)
+    label_color_index = {}
+    for label, color_str in color_dict.items():
+        color = Color(transform_color(color_str)[0])
+        for i in range(len(colormap.colors)):
+            if colormap.colors[i] == color:
+                label_color_index[label] = colormap._controls[i]
```

Comment:
```suggestion
    for i, (label, color_str) in enumerate(color_dict.items()):
        label_color_index[label] = i / (len(colors) - 1)
```

The Colormap deterministically sets control points based on the order of the colors passed to it, evenly spacing them out between 0.0 and 1.0 so I don't think you actually need to do that second search through the colormap to find the color. I just tried the above code and it works fine (produces an identical dictionary up to floating point precision). Probably simpler just to have this single pass

Reply:
I see, I was a bit scared to make this assumption, but as long as it works :)